src/main/java/fr/igred/omero/repository/ImageWrapper.java

```diff
@@ -618,4 +617,31 @@ public List<File> download(Client client, String path)
         return new ArrayList<>(0);
     }
 
+
+    /**
+     * Get the parent dataset information of an image.
+     *
+     * @param client The client handling the connection.
+     * @param imageId The image ID coming from OMERO
+     *
+     * @return see above
+     *
+     * @throws DSOutOfServiceException
+     * @throws ServerError
+     */
+    public ArrayList<DatasetData> getParentDatasets(Client client, Long imageId) throws DSOutOfServiceException, ServerError {
```

Comment:
Returns `ArrayList<DatasetData>` and throws `DSOutOfServiceException` as well as `ServerError`, which forces downstream to include OMERO Gateway dependencies